stdlib.sh

```diff
@@ -1068,6 +1068,34 @@ layout_pyenv() {
   [[ -n "$PYENV_VERSION" ]] && export PYENV_VERSION
 }
 
+# Usage: layout uv
+#
+# Enables the uv project layout in the current directory, and syncs
+# the dependencies in the project.
+#
+# This relies on the `uv` command being available in the PATH, and performs a
+# sync on cd because uv is fast enough it's not impactful. It relies on uv's
+# configuration file and environment variables, rather than arguments.
+#
+layout_uv() {
+  # Watch the uv configuration file for changes
+  watch_file .python-version pyproject.toml uv.lock
+
+  # Use --frozen so that direnv entry does not change the lock contents. This
+  # can fail if the lock file is badly out of sync, but it'll report that to the
+  # user and the resolution is obvious.
+  uv sync --frozen || true
+
+  # activate the virtualenv after syncing; this puts the newly-installed
+  # binaries on PATH.
+  venv_path="$(expand_path "${UV_PROJECT_ENVIRONMENT:-.venv}")"
+  if [[ -f $venv_path ]]; then
+    # shellcheck source=/dev/null
+    source "$venv_path/bin/activate"
+  fi
```

Comment:
I found a cleaner way to do the venv activation in [the doc](https://direnv.net/man/direnv-stdlib.1.html#codedirenvload-ltcommand-generating-dump-outputgtcode):

```suggestion
  direnv_load uv run --frozen direnv dump
```

Tried locally and it seems to work. It's more compact and it saves us from reimplementing the `UV_PROJECT_ENVIRONMENT` logic, also I guess it addresses the cross-platform concerns.

And btw `uv run` implicitly calls `uv sync`. We can pass `--no-sync` to avoid that, or `--locked`/`--frozen`. So yeah it's a two-liner in the end 😁 

(I didn't check what's the result in case of network/other errors. Maybe it's still better to split the `sync --frozen` and the `run --no-sync` to still have something to work with)

Reply:
What is the goal here? What does this give us over simply activating the virtualenv?

Reply:
There is one difference between the two; `uv` doesn't set `VIRTUAL_ENV_PROMPT` in the same way that the `.venv/bin/activate` does. 

At the end of the day, the difference is slim enough that I'm not super worried about it. If using the `direnv_load` approach is a blocker to merging, then I'll switch to that. Is it?

Reply:
No idea. I was just reading the doc and it looked like a good idea

Reply:
I notice that the `uv run --no-sync` approach works *even with no venv*, which makes it the purest "just set environment variables" option that I've seen. Specifically, I can use `direnv_load uv run --no-sync direnv dump`, no filesystem modifications are performed, I can run `uv sync` afterwards, and everything seems to work normally.